specification/workloads/resource-manager/Microsoft.Workloads/connectors/preview/2023-10-01-preview/connectors.json

```diff
@@ -0,0 +1,1098 @@
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ConnectorProperties": {
+      "description": "Defines the connector resource properties.",
+      "type": "object",
+      "required": [
+        "sourceResourceId"
+      ],
+      "properties": {
+        "sourceResourceId": {
+          "description": "Defines the ID of the connector's source resource.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ConnectorProvisioningState",
+          "readOnly": true
+        },
+        "errors": {
+          "description": "Indicates any errors on the connector resource.",
+          "$ref": "#/definitions/ConnectorErrorDefinition",
+          "readOnly": true
+        },
+        "managedResourceGroupConfiguration": {
+          "description": "Managed resource group configuration",
+          "$ref": "#/definitions/ManagedRGConfiguration"
+        }
+      }
+    },
+    "Connector": {
+      "description": "Define the connector resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "$ref": "../../../common-types/v1/commonTypes.json#/definitions/UserAssignedServiceIdentity"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ConnectorProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "ConnectorList": {
+      "description": "Defines the collection of connector resources.",
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "description": "Gets the list of connector resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Connector"
+          }
+        },
+        "nextLink": {
+          "description": "Gets the value of next link.",
+          "type": "string",
+          "format": "uri"
+        }
+      }
+    },
+    "ACSSBackupConnectionList": {
+      "description": "Defines the collection of backup connection resources of virtual instances for SAP.",
+      "type": "object",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "description": "Gets the list of backup connection resources of virtual instances for SAP.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ACSSBackupConnection"
+          }
+        },
+        "nextLink": {
+          "description": "Gets the value of next link.",
+          "type": "string",
+          "format": "uri"
+        }
+      }
+    },
+    "NewRecoveryServicesVault": {
+      "x-ms-discriminator-value": "New",
```

Comment:
>New [](http://example.com/codeflow?start=35&length=3)

Please do review some of the generated SDKs. I believe the discriminators are resolved globally and not within the context of local types. This would imply a resource with discriminator value of New would conflict with any other resource definition that uses the discriminator value of New, even if it is a different type.